webhonc/src/index.ts

```diff
@@ -0,0 +1,94 @@
+import { zValidator } from "@hono/zod-validator";
+import { Hono } from "hono";
+import { z } from "zod";
+import type { Bindings, Variables } from "./types";
+import { WebHonc } from "./webhonc";
+
+const app = new Hono<{ Bindings: Bindings; Variables: Variables }>();
+
+app.get("/ws", async (c) => {
+  if (c.req.header("upgrade") !== "websocket") {
+    return new Response("Not a websocket request", { status: 426 });
+  }
+
+  const id = c.env.WEBHONC.newUniqueId();
+  const webhonc = c.env.WEBHONC.get(id) as DurableObjectStub<WebHonc>;
+
+  return webhonc.fetch(c.req.raw);
+});
+
+app.all(
+  "/:id/*",
+  zValidator("param", z.object({ id: z.string() })),
+  async (c) => {
+    const { id } = c.req.valid("param");
+
+    // We need to serialize the path components without the id
+    // so we can send it down the wire and replay it locally
+    const pathComponentsWithoutId = c.req.path
+      .split("/")
+      .filter((comp) => comp.length > 0)
+      .slice(1);
+
+    const method = c.req.method;
+
+    const contentType = c.req.header("content-type");
+
+    let body: string | FormData | null;
+
+    // if the request is a GET or HEAD, we don't want to send the body
+    if (method === "GET" || method === "HEAD") {
+      body = null;
+    } else {
+      switch (contentType) {
+        case "application/json":
+          body = await c.req.json();
+          break;
+        case "application/x-www-form-urlencoded":
+          body = await c.req.formData();
+          break;
+        case "text/plain":
+          body = await c.req.text();
+          break;
+        default:
```

Comment:
additional cases i think we should cover:

- `text/html`
- `application/xml`

Reply:
html is a little ridiculous but ya know. doesn't hurt to support it. just need to parse the body as text for both of those cases

Reply:
handled (even csv's now)